### ISSUE TITLE
Pyright thinks `typing.Union` is `object` instead of `type`

### ISSUE THREAD

**Issue by CobaltCause** [microsoft/pyright#3522]
**Describe the bug**

Pyright thinks `typing.Union` is an `object` and therefore can't be assigned to `type`. Maybe it's right, but this seems surprising.

In particular, I'm running into [this issue with FastAPI](https://github.com/tiangolo/fastapi/issues/2279). I tried the `TypeVar` workaround suggested there and that doesn't work either (Pyright says "`TypeVar` cannot be assigned to `type`").

**To Reproduce**

This code:

```python
from typing import Any, Tuple, Type, Union


def takes_type(_: Type[Any]):
    pass


takes_type(Tuple[str, int])
takes_type(Union[str, int])
```

Makes Pyright say:

```
error: Argument of type "Type[str] | Type[int]" cannot be assigned to parameter "_" of type "type" in function "takes_type"
    "object" is incompatible with "type" (reportGeneralTypeIssues)
```

**Expected behavior**

The code should typecheck.

**Additional context**

I am using Python 3.10 and running Pyright 1.1.249 at the command line.

**Comment by erictraut** [comment]
Yes, this behavior is correct and intended. `Union` is a special form that operates on types. It is not a type itself. Refer to [PEP 484](https://peps.python.org/pep-0484/#the-type-of-class-objects) for details. Other type checkers like mypy also emit the same error in this case.
